### DISCUSSION TITLE
Release 5.0.1

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#1069] [Announcement]
Release 5.0.1
===

The main reason for this release is to add an 'odyssey' boolean flag to all EDDN messages for the benefit of listeners, e.g. eddb.io, inara.cz, edsm.net, spansh.co.uk, etc.  **Please do update so as to make their lives easier once Odyssey has launched!**

* Translations have been updated again.  Thanks to all the contributors.  See [wiki:Translations](https://github.com/EDCD/EDMarketConnector/wiki/Translations) and [Translations welcome](https://github.com/EDCD/EDMarketConnector/issues/24) for links and discussion if you want to help.

* Changed the error message "`Error: Frontier server is down`" to "`Error: Frontier CAPI didn't respond`" to make it clear this pertains to the CAPI and not the game servers.

Killswitches
---

In the 5.0.0 changelog we said:

  <blockquote>We will **NOT** be using this merely to try and get some laggards to upgrade.</blockquote>

However, from now on there is an exception to this.  **After** this release any subsequent -beta or -rc versions will be killswitched *after* their full release is published.

For example, if we put out a `5.0.2-beta1` and `5.0.2-rc1` before the full `5.0.2`, then when `5.0.2` was published we would activate all available killswitches for versions `5.0.2-beta1` and `5.0.2-rc1`.  In this example `5.0.1` would **not** be killswitched as part of *this policy* (but still could be if, e.g. a data corruption bug was found in it).

In general please do **not** linger on any -beta or -rc release if there has been a subsequent release.  Upgrade to the equivalent full release once it is published.

Plugin Developers
---

* Please make the effort to subscribe to GitHub notifications of new EDMarketConnector releases:

  1. Login to [GitHub](https://github.com).
  2. Navigate to [EDMarketConnector](https://github.com/EDCD/EDMarketConnector).
  3. Click the 'Watch' (or 'Unwatch' if you previously set up any watches on us).  It's currently (2021-05-13) the left-most button of 3 near the top-right of the page.
  4. Click 'Custom'.
  5. Ensure 'Releases' is selected.
  6. Click 'Apply'.

  This way you'll be aware, as early as possible, of any -beta and -rc changelogs and changes that might affect your work.

* `state` passed to `journal_entry()` has a new member `Odyssey` (note the capital `O`) which is a boolean indicating if the `LoadGame` event both has an `Odyssey` key, and if so, what the value was.  Defaults to `False`.

* PLUGINS.md updated to document the `state['Horizons']` flag that has been present in it since version 3.0 of the game.

* The `stations.p` and `systems.p` files that were deprecated in 5.0.0 have now also been removed in git.  As this release is made they will no longer be in the `develop`, `main` or `stable` branches.  If you truly need to find a copy look at the `Release/4.2.7` tag, but do read the 5.0.0 changelog for why we stopped using them and what you can change to also not need them.

<hr /><em>This discussion was created from the release <a href='https://github.com/EDCD/EDMarketConnector/releases/tag/Release/5.0.1'>Release 5.0.1</a>.</em>

**Comment by Athanasius** [comment]
If anyone is wondering why 5.0.0 didn't contain this 'odyssey' flag code, it was released before the decision was made to add that flag to the EDDN schemas.  Sending it before the EDDN Gateway had the new schemas loaded would have resulted in messages being rejected.

**Comment by Banzai51** [comment]
Update to 5.0.1 and lost all my settings.

**Reply by Athanasius** [reply]
Please open [a bug report](https://github.com/EDCD/EDMarketConnector/issues/new?assignees=&labels=bug%2C+unconfirmed&template=bug_report.md&title=) about that.

Specifically we need to know:

1. What version of EDMC you had before this.
2. Are you sure this isn't just a case of your last game session being a quick "start the game, get to main menu, exit game", see https://github.com/EDCD/EDMarketConnector/wiki/Troubleshooting#credentials-settings-are-greyed-out
2. Definitely both log files
3. Do you have more than one game account ?
4. Which Commander did you play most recently ?

**Reply by Banzai51** [reply]
Rebooted and fired up E:D, then the settings returned. In the state it was, I couldn't even edit the info. But back to normal.

**Reply by Athanasius** [reply]
Yes, that's EXACTLY the scenario outlined in that Troubleshooting entry.  Glad it's not an actual bug :)

**Comment by Athanasius** [comment]
Locking this conversation as 5.0.2 will be out shortly with its own thread.
